infra/app/service/image_tag.tf

```diff
@@ -6,6 +6,15 @@
 #  1. Accept an optional variable during a terraform plan/apply. (see "image_tag" variable in variables.tf)
 
 #  2. Read the output used from the last terraform state using "terraform_remote_state".
+#     Get the backend config by parsing the backend config file
+locals {
+  backend_config_file_path = "${path.module}/${var.environment_name}.s3.tfbackend"
+  backend_config_file      = file("${path.module}/${var.environment_name}.s3.tfbackend")
+  backend_config_regex     = "(\\w+)\\s*=\\s*\\\"(\\w+)\\\""
+  backend_config           = { for match in regexall("(\\w+)\\s*=\\s*\\\"(.+)\\\"", local.backend_config_file) : match[0] => match[1] }
```

Comment:
👀 👀 These regexes. Can they be avoided at all? If no, can they be documented?

Reply:
yeah i'll add a comment.
the regex itself isn't that bad, but terraform unfortunately doesn't have native way to write regexes (unlike in javascript or python), so we have to double escape them. the regex is actually just:

```
(\w+)\s*="\s*(.+)"
```

also i realized i typed the regex twice accidentally and i overescaped the quotes

Reply:
re: whether we can avoid them: 
if you know of a native terraform function that can parse tfvars files that'd be great, but i couldn't find one

Reply:
Ok simplified the regex and added a comment. Do you think that makes it clearer?

Reply:
Yup. Looks great! Thanks